pkg/src/src/main/scala/edu/berkeley/cs/amplab/sparkr/SQLUtils.scala

```diff
@@ -1,10 +1,74 @@
 package edu.berkeley.cs.amplab.sparkr
 
 import org.apache.spark.SparkContext
-import org.apache.spark.sql.{SQLContext}
+import org.apache.spark.rdd.RDD
+import org.apache.spark.api.java.JavaRDD
+import org.apache.spark.sql.{SQLContext, DataFrame, Row}
+import org.apache.spark.sql.types.{StructType}
+
+import edu.berkeley.cs.amplab.sparkr.SerDe._
+
+import java.io.ByteArrayOutputStream
+import java.io.DataOutputStream
 
 object SQLUtils {
   def createSQLContext(sc: SparkContext): SQLContext = {
     new SQLContext(sc)
   }
+
+  def getColNames(df: DataFrame): Array[String] = {
+    df.schema.fields.map(_.name)
+  }
+
+  def dfToRowRDD(df: DataFrame): JavaRDD[Array[Byte]] = {
+    df.map(r => rowToRBytes(r))
+  }
+
+  def rowToRBytes(row: Row): Array[Byte] = {
```

Comment:
Could you mark this as `private[this] def rowToRBytes(row: Row)...`? It is only used once inside this object.